### PR TITLE
Add Command environment variable inheritance docs

### PR DESCRIPTION
**Pull request by schneems**
The interaction between the environment variable methods can be confusing. Specifically `env_clear` and `remove_env` have a side effects not mentioned: they disable inheriting environment variables from the parent process. I wanted to fully document this behavior as well as explain relevant edge cases in each of the `Command` env methods.

This is further confused by the return of `get_envs` which will return key/None if `remove_env` has been used, but an empty iterator if `env_clear` has been called. Or a non-empty iterator if `env_clear` was called and later explicit mappings are added. Currently there is no way (that I'm able to find) of observing whether or not the internal `env_clear=true` been toggled on the `Command` struct via its public API.

Ultimately environment variable mappings can be in one of several states:

- Explicitly set value (via `envs` / `env`) will take precedence over parent mapping
- Not explicitly set, will inherit mapping from parent
- Explicitly removed via `remove_env`, this single mapping will not inherit from parent
- Implicitly removed via `env_clear`, no mappings will inherit from parent

I tried to represent this in the relevant sections of the docs. 

This is my second-ever doc PR (whoop!). I'm happy to take specific or general doc feedback. Also happy to explain the logic behind any changes or additions I made.

### CODE REVIEW

File: library/std/src/process.rs [hunk 1]
```diff
@@ -980,17 +1022,21 @@ impl Command {
...
+    /// [`Command::env_remove`] can be retrieved with this method.
+    ///
+    /// Note that this output does not include environment variables inherited from the parent
+    /// process.
     ///
-    /// Each element is a tuple `(&OsStr, Option<&OsStr>)`, where the first
-    /// value is the key, and the second is the value, which is [`None`] if
-    /// the environment variable is to be explicitly removed.
+    /// Each element is a tuple key/value pair `(&OsStr, Option<&OsStr>)`. A [`None`] value
+    /// indicates its key was explicitly removed via [`Command::env_clear`]. The associated key for
```
Comment (cuviper) [inline library/std/src/process.rs:1034]: I think you meant:
```suggestion
    /// indicates its key was explicitly removed via [`Command::env_remove`]. The associated key for
```

**Comment by cuviper** [comment]
AFAICT you've accurately captured the current behavior, but I'm going to flip this over to API review, since documentation implies a stronger promise about maintaining that exactly.

r? libs-api

**Comment by Amanieu** [comment]
r=me once @cuviper's comment is addressed

**Comment by schneems** [comment]
Sorry for the delay. I applied @cuviper's suggestion, it was a good catch. Thanks for the reviews and all your work on rust ❤️ 

Let me know if this needs anything else.

**Comment by schneems** [comment]
> r=me once @cuviper's comment is addressed

I didn't realize this was shorthand for literal syntax. But now I see it above. Not sure If I can trigger it but trying now:

r? @Amanieu

**Comment by cuviper** [comment]
The intent of saying "`r=me` when..." is that anyone with review privileges can submit it on their behalf. But somebody also has to notice that it's ready, sorry about that! So...

@bors r=Amanieu

**Pull request merged**